src/linear_algebra/matrix.lean

```diff
@@ -936,3 +936,35 @@ def alg_equiv_matrix {R : Type v} {M : Type w} {n : Type*} [fintype n]
   [comm_ring R] [add_comm_group M] [module R M] [decidable_eq n] {b : n → M} (h : is_basis R b) :
   module.End R M ≃ₐ[R] matrix n n R :=
 h.equiv_fun.alg_conj.trans alg_equiv_matrix'
+
+section
+
+variables {R : Type v} [semiring R] {n : Type w} [fintype n]
+
+@[simp] lemma matrix.dot_product_std_basis_one [decidable_eq n] (v : n → R) (i : n) :
+  matrix.dot_product v (linear_map.std_basis R (λ _, R) i 1) = v i :=
```

Comment:
It's tempting to add another simp lemma that says
```
  matrix.dot_product v (linear_map.std_basis R (λ _, R) i c) = c \smul v i :=
```
although obviously that's only in scope for this PR if you want it to be :)

Reply:
is `smul` the same as multiplication on semirings?

Reply:
We don't have right-actions at the moment, I don't think. Good point though.

If you swapped the argument order my proposed lemma would hold

Reply:
I have `matrix.dot_product v (linear_map.std_basis R (λ _, R) i c) = v i * c :=` working if that's what you have in mind